content/code-signing/macos-code-signing.md

```diff
@@ -0,0 +1,128 @@
+---
+description: How to set up macOS code signing in the Flutter workflow editor
+title: macOS code signing
+weight: 2
+---
+
+Code signing is required by Apple for integrating app services, installing your macOS app on another machine or for uploading it to distribute it through Mac App Store or outside of Mac App Store. It enables to identify who developed the app and ensure that all the changes to the app come from you or your team.
+
+To create an application package which can be published to Mac App Store on Codemagic, you need to set up code signing.
+
+{{<notebox>}}
+This guide only applies to workflows configured with the **Flutter workflow editor**. If your workflow is configured with **codemagic.yaml** please go to [Signing macOS apps using codemagic.yaml](../code-signing-yaml/signing-macos).
+{{</notebox>}}
+
+## Prerequisites
+
+Before you can start signing or distributing your app, there are a few things you need.
+
+* [Apple Developer Program](https://developer.apple.com/programs/enroll/) membership
+* An [app ID](https://developer.apple.com/account/resources/identifiers/bundleId/add/) for your macOS app in Apple Developer portal
+* **signing certificates** (`Personal Information Exchange, .p12`)
+* A **provisioning profile** (`.provisionprofile`)
+
+The signing certificates — development or distribution — help to identify who built the code.
+
+A provisioning profile — development or distribution — contains information about the app ID, the devices on which the app can be installed and the certificates that can be used for signing the app. Note that if your app contains app extensions, you need an additional provisioning profile for each app extension.
+
+{{<notebox >}}
+With **automatic code signing**, Codemagic will create both the certificate and the provisioning profile for you on your behalf without requiring a Mac.
+
+With **manual code signing**, you need to upload the signing files manually.
+{{</notebox>}}
+
+Signing an application with a development certificate and profile requires the UUID of the machine which does the build to be present in the profile. Therefore, it's not possible to do with Codemagic since the build machine won't be listed in the used profile.
+
+Distibution is possible to Mac App Store (using `Mac App Distribution` and `Mac Installer Distribution` certificates and `Mac App Store` profile) and outside of Mac App Store (using `Developer ID Application` certificate and `Developer ID` profile). **Currently Codemagic UI only supports publishing to Mac App Store.**
```

Comment:
Random question, could we support code signing to not the mac app store in the future? Could we do this in codemagic.yaml?

Reply:
yes, sure, i already built outside of mac app store signing apps with codemagic yaml. it will require some development for UI, but for yaml it's easier. It's harder to get an app compliant with notarization requirements than to do code signing:)

Reply:
Seems like something to direct to codemagic.yaml on a case by case basis. :+1: